miasm2/jitter/arch/JitCore.h

```diff
@@ -54,20 +60,75 @@
 #define getset_reg_u16(regname)						\
 	static PyObject *JitCpu_get_ ## regname  (JitCpu *self, void *closure) \
 	{								\
-		return PyLong_FromUnsignedLongLong((uint16_t)(self->vmcpu.  regname  )); \
+		return PyLong_FromUnsignedLongLong((uint16_t)(((vm_cpu_t*)(self->cpu))-> regname  )); \
 	}								\
 	static int JitCpu_set_ ## regname  (JitCpu *self, PyObject *value, void *closure) \
 	{								\
 		uint16_t val;						\
 		PyGetInt_ret0(value, val);				\
-		self->vmcpu.  regname   = val;				\
+		((vm_cpu_t*)(self->cpu))->  regname   = val;				\
 		return 0;						\
 	}
 
 
+#define get_reg(reg)  do {						\
+		o = PyLong_FromUnsignedLongLong((uint64_t)((vm_cpu_t*)(self->cpu))->reg); \
+		PyDict_SetItemString(dict, #reg, o);			\
+		Py_DECREF(o);						\
+	} while(0);
+
+
+#define get_reg_off(reg)  do {						\
+		o = PyLong_FromUnsignedLongLong((uint64_t)offsetof(vm_cpu_t, reg)); \
+		PyDict_SetItemString(dict, #reg, o);			\
+		Py_DECREF(o);						\
+	} while(0);
+
+
 typedef struct {
 	uint8_t is_local;
 	uint64_t address;
 } block_id;
 
+typedef struct {
+	PyObject_HEAD
+	PyObject *pyvm;
+	PyObject *jitter;
+	void* cpu;
+} JitCpu;
+
+
+typedef struct _reg_dict{
+    char* name;
+    size_t offset;
+} reg_dict;
+
+
+
+void JitCpu_dealloc(JitCpu* self);
+PyObject * JitCpu_new(PyTypeObject *type, PyObject *args, PyObject *kwds);
+PyObject * JitCpu_get_vmmngr(JitCpu *self, void *closure);
+PyObject * JitCpu_set_vmmngr(JitCpu *self, PyObject *value, void *closure);
+PyObject * JitCpu_get_jitter(JitCpu *self, void *closure);
+PyObject * JitCpu_set_jitter(JitCpu *self, PyObject *value, void *closure);
 void Resolve_dst(block_id* BlockDst, uint64_t addr, uint64_t is_local);
+
+
+
+uint8_t MEM_LOOKUP_08(JitCpu* jitcpu, uint64_t addr);
+uint16_t MEM_LOOKUP_16(JitCpu* jitcpu, uint64_t addr);
+uint32_t MEM_LOOKUP_32(JitCpu* jitcpu, uint64_t addr);
+uint64_t MEM_LOOKUP_64(JitCpu* jitcpu, uint64_t addr);
+void MEM_WRITE_08(JitCpu* jitcpu, uint64_t addr, uint8_t src);
+void MEM_WRITE_16(JitCpu* jitcpu, uint64_t addr, uint16_t src);
+void MEM_WRITE_32(JitCpu* jitcpu, uint64_t addr, uint32_t src);
+void MEM_WRITE_64(JitCpu* jitcpu, uint64_t addr, uint64_t src);
+PyObject* __attribute__((weak)) vm_get_mem(JitCpu *self, PyObject* args);
```

Comment:
Why do you repeat the `__attribute__((weak))` for `vm_get_mem` but not for others functions such as `MEM_WRITE_*` ?